EnrichStream/Dockerfile

```diff
@@ -1,58 +1,38 @@
 #
-#  Install dependencies, compile sources, build jars.
+#  Install dependencies, compile sources, build jar.
 #
 FROM maven:3.9.6-eclipse-temurin-17-alpine as build
 
 WORKDIR /app
 
-ARG MAVEN_ARGS
-
 COPY pom.xml ./
 COPY settings.xml ./
-# Copy in local repository
-COPY .m2/ /root/.m2/
+COPY .mvn .mvn
 
 # Download dependencies (with docker cache)
-RUN \
-    --mount=type=secret,id=GITHUB_TOKEN \
+RUN --mount=type=secret,id=GITHUB_TOKEN \
     --mount=type=secret,id=GITHUB_ACTOR \
-    --mount=type=cache,target=./.m2,sharing=locked \
-    cp -rn /root/.m2/ ./ &&\
+    --mount=type=cache,target=/root/.m2,sharing=locked \
```

Comment:
BEFORE:
copy local .m2 into docker layer
copy to /root/.m2 in docker layer
restore from /root/.m2

NOW:
run maven and assume .m2 is inside /root/.m2 and use docker layer caching

Are we sure it works? and wont this make docker build locally slower, as it won't copy your local cache into docker.

Reply:
before, the dependencies would be cached separately for each project since each project has its own .m2 directory inside the project directory. this means no sharing of the cached dependencies across different projects. the only advantage of this that i can think of is that it is possible to copy from your own local .m2 directory into the .m2 directory for the project, but this seems hacky and i dont think it should be used for anything but debugging

when using /root/.m2, docker will still not be able to cache across different projects, however, when running maven on its own, the caching will work across different projects. it avoids some pretty annoying dependency juggling and relies on maven standards which i much prefer